### PR TITLE
use react to create a portal so that context is available

### PR DESCRIPTION
**Pull request by jhsu**
lucid's portal implementation loses react context.

This pull request changes the portal implementation to use `React.createPortal` to [ensure that context is preserved](https://reactjs.org/docs/portals.html).

An example using context has been added.

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [ ] Edge (Win 10)
- [ ] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- [ ] One core team UX approval

### CODE REVIEW

**Comment by astropunk** [comment]
thanks joe! 👏

File: src/components/Portal/Portal.jsx [hunk 1]
```diff
@@ -44,47 +26,55 @@ const Portal = createClass({
...
+			document.body.appendChild(portalElement);
 		}
 		this.portalElement = portalElement;
-		this.componentDidUpdate();
+		this.setState({ isReady: true });
 	},
 	componentWillUnmount() {
-		ReactDOM.unmountComponentAtNode(this.portalElement);
-		window.document.body.removeChild(this.portalElement);
+		if (this.manuallyCreatedPOrtal) {
```
Comment (jondlm) [inline src/components/Portal/Portal.jsx]: `manuallyCreatedPOrtal` looks misspelled

Comment (jhsu) [reply src/components/Portal/Portal.jsx]: whoops, will fix this

**Comment by jhsu** [comment]
@jondlm yeah i poked around at things using Dialog/Overlay/Portal

File: src/components/Portal/Portal.jsx (hunk 1 above)
Comment (jhsu) [reply src/components/Portal/Portal.jsx]: fixed and added a test case

**Pull request merged**